pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -112,14 +103,57 @@ func MergeRuntimeContainers(runtimeContainer *v1.Container, predictorContainer *
 		coreContainer.Name = runtimeContainerName
 	}
 
-	argCopy = append(argCopy, predictorContainer.Args...)
-	envCopy = append(envCopy, predictorContainer.Env...)
-
-	coreContainer.Args = argCopy
-	coreContainer.Env = envCopy
+	// Args and Env will be combined instead of overridden
+	argCopy := make([]string, len(runtimeContainer.Args))
+	copy(argCopy, runtimeContainer.Args)
+	coreContainer.Args = append(argCopy, predictorContainer.Args...)
+	coreContainer.Env = MergeEnvs(runtimeContainer.Env, predictorContainer.Env)
 
 	return &coreContainer, nil
+}
 
+// MergeRuntimeVolumes Merge the predictor Volumes list with the runtime Volumes list, allowing users
+// to override/augment runtime volume settings from the predictor spec.
+func MergeRuntimeVolumes(runtimeVolumes []v1.Volume, predictorVolumes []v1.Volume) []v1.Volume {
+	if len(runtimeVolumes) == 0 {
+		return predictorVolumes
+	} else if len(predictorVolumes) == 0 {
+		return runtimeVolumes
+	}
+	vols := append(([]v1.Volume)(nil), runtimeVolumes...)
```

Comment:
If you want to ensure a single allocation, can extend the slice with capacity in case each predictor entry is added:
```suggestion
	vols := make([]v1.Volume, len(runtimeVolumes), len(runtimeVolumes)+len(predictorVolumes))
	copy(vols, runtimeVolumes)
```

Reply:
I guess I wasn't too concerned about the allocation here given how infrequently this will run and likelihood that the slices are small, so opted to keep concise.